components/renku_data_services/notebooks/api/schemas/cloud_storage.py

```diff
@@ -57,54 +56,55 @@ def __init__(
         readonly: bool,
         mount_folder: str,
         name: Optional[str],
-        config: NotebooksConfig,
+        secrets: dict[str, str],  # "Mapping between secret ID (key) and secret name (value)
+        storage_class: str,
+        user_secret_key: str | None = None,
     ) -> None:
         """Creates a cloud storage instance without validating the configuration."""
-        self.config = config
         self.configuration = configuration
         self.source_path = source_path
         self.mount_folder = mount_folder
         self.readonly = readonly
         self.name = name
+        self.secrets = secrets
+        self.base_name: str | None = None
+        self.user_secret_key = user_secret_key
+        self.storage_class = storage_class
 
     @classmethod
     async def storage_from_schema(
         cls,
         data: dict[str, Any],
-        user: APIUser,
-        internal_gitlab_user: APIUser,
-        project_id: int,
         work_dir: PurePosixPath,
-        config: NotebooksConfig,
+        saved_storage: CloudStorage | None,
+        storage_class: str,
+        user_secret_key: str | None = None,
     ) -> Self:
         """Create storage object from request."""
         name = None
-        if data.get("storage_id"):
-            # Load from storage service
-            if user.access_token is None:
-                raise ValidationError("Storage mounting is only supported for logged-in users.")
-            if project_id < 1:
-                raise ValidationError("Could not get gitlab project id")
-            (
-                configuration,
-                source_path,
-                target_path,
-                readonly,
-                name,
-            ) = await config.storage_validator.get_storage_by_id(
-                user, internal_gitlab_user, project_id, data["storage_id"]
-            )
-            configuration = {**configuration, **(data.get("configuration", {}))}
-            readonly = readonly
+        if saved_storage:
+            configuration = {**saved_storage.configuration.model_dump(), **(data.get("configuration", {}))}
+            readonly = saved_storage.readonly
+            name = saved_storage.name
         else:
             source_path = data["source_path"]
             target_path = data["target_path"]
             configuration = data["configuration"]
             readonly = data.get("readonly", True)
-        mount_folder = str(work_dir / target_path)
 
-        await config.storage_validator.validate_storage_configuration(configuration, source_path)
-        return cls(source_path, configuration, readonly, mount_folder, name, config)
+        # NOTE: This is used only in Renku v1, there we cannot save secrets for storage
```

Comment:
```suggestion
        # NOTE: This is used only in Renku v1, there we do not save secrets for storage
```

We could save them but we don't.